grafana/resource_cloud_stack.go

```diff
@@ -268,3 +287,28 @@ func FlattenStack(d *schema.ResourceData, stack gapi.Stack) {
 	d.Set("alertmanager_url", stack.AmInstanceURL)
 	d.Set("alertmanager_status", stack.AmInstanceStatus)
 }
+
+// waitForStackReadiness retries until the stack is ready, verified by querying the Grafana URL
+func waitForStackReadiness(ctx context.Context, d *schema.ResourceData) diag.Diagnostics {
+	if wait := d.Get("wait_for_readiness").(bool); !wait {
+		return nil
+	}
+
+	err := resource.RetryContext(ctx, 2*time.Minute, func() *resource.RetryError {
+		resp, err := http.Head(d.Get("url").(string))
+		if err != nil {
+			return resource.NonRetryableError(err)
+		}
+		defer resp.Body.Close()
+		if resp.StatusCode != 200 {
+			return resource.RetryableError(errors.New("stack is not ready yet"))
+		}
+
+		return nil
+	})
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error waiting for stack to be ready: %s", err))
```

Comment:
I haven't used the diag errors, but isn't `%w` always more useful?
```suggestion
		return diag.FromErr(fmt.Errorf("error waiting for stack to be ready: %w", err))
```